plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConnector.java

```diff
@@ -0,0 +1,150 @@
+import io.trino.spi.connector.ConnectorSplitManager;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.session.PropertyMetadata;
+import io.trino.spi.transaction.IsolationLevel;
+import jakarta.inject.Inject;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.spi.connector.ConnectorCapabilities.NOT_NULL_COLUMN_CONSTRAINT;
+import static io.trino.spi.session.PropertyMetadata.doubleProperty;
+import static io.trino.spi.session.PropertyMetadata.longProperty;
+import static io.trino.spi.session.PropertyMetadata.stringProperty;
+import static java.util.Objects.requireNonNull;
+
+public class FakerConnector
+        implements Connector
+{
+    private final FakerMetadata metadata;
+    private final FakerSplitManager splitManager;
+    private final FakerPageSourceProvider pageSourceProvider;
+    private final FakerPageSinkProvider pageSinkProvider;
+
+    @Inject
+    public FakerConnector(
+            FakerMetadata metadata,
+            FakerSplitManager splitManager,
+            FakerPageSourceProvider pageSourceProvider,
+            FakerPageSinkProvider pageSinkProvider)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.pageSourceProvider = requireNonNull(pageSourceProvider, "pageSourceProvider is null");
+        this.pageSinkProvider = requireNonNull(pageSinkProvider, "pageSinkProvider is null");
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly, boolean autoCommit)
+    {
+        return FakerTransactionHandle.INSTANCE;
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata(ConnectorSession session, ConnectorTransactionHandle transaction)
+    {
+        return metadata;
+    }
+
+    @Override
+    public ConnectorSplitManager getSplitManager()
+    {
+        return splitManager;
+    }
+
+    @Override
+    public ConnectorPageSourceProvider getPageSourceProvider()
+    {
+        return pageSourceProvider;
+    }
+
+    @Override
+    public ConnectorPageSinkProvider getPageSinkProvider()
+    {
+        return pageSinkProvider;
+    }
+
+    @Override
+    public Set<ConnectorCapabilities> getCapabilities()
+    {
+        return Set.of(NOT_NULL_COLUMN_CONSTRAINT);
+    }
+
+    @Override
+    public List<PropertyMetadata<?>> getSchemaProperties()
+    {
+        return ImmutableList.of(
+                doubleProperty(
+                        SchemaInfo.NULL_PROBABILITY_PROPERTY,
+                        "Default probability of null values in any column that allows them, in any table of this schema",
+                        null,
+                        nullProbability -> checkState(0 <= nullProbability && nullProbability <= 1, "null_probability value must be between 0 and 1, inclusive"),
```

Comment:
checkState is not the right exception here, we need to throw `io.trino.spi.StandardErrorCode#INVALID_SCHEMA_PROPERTY`